### DISCUSSION TITLE
Is there any established way to script/manipulate the simulator?

### DISCUSSION THREAD

**Discussion by Triq1** [ricktu288/ray-optics#143] [Q&A] [answered]
I am trying to simulate a 2d "missile seeker" (a lens with two photodiodes) to allow the vehicle to track and aim toward the light source. To do this, I need to be able to move and aim the "sensing system" (the lens and detector assembly) and iterate over time with a "guidance algorithm" (a little py script).

**Comment by ricktu288** [comment, marked as answer]
The only way available currently is to use the JavaScript console of the browser. See #142.

**Comment by ricktu288** [comment]
See https://github.com/ricktu288/ray-optics/discussions/141#discussioncomment-9515057 for a new experimental implementation of module object. Currently there is no iterating feature but may be added later.

**Reply by Triq1** [reply]
Seems interesting, I'll take a look!
